include/fakeit/StubbingProgress.hpp

```diff
@@ -50,7 +50,17 @@ namespace fakeit {
         template<typename U = R>
         typename std::enable_if<!std::is_reference<U>::value, MethodStubbingProgress<R, arglist...> &>::type
         Return(const R &r) {
-            return Do([r](const typename fakeit::test_arg<arglist>::type...) -> R { return r; });
+            return Do([r](const typename fakeit::test_arg<arglist>::type...) mutable -> R { return r; });
+        }
+
+        template<typename T>
+        typename std::enable_if<!std::is_reference<T>::value && std::is_copy_constructible<T>::value, MethodStubbingProgress<R, arglist...>&>::type
+    	Return(T&& t) {
```

Comment:
Currently I'm getting failures for this line/test: cannot convert argument 1 from 'int' to 'int &'

```
 			When(Method(mock, returnIntByRef)).Return(1);
```

```suggestion
        template<typename U = R>
        typename std::enable_if<!std::is_reference<T>::value && std::is_copy_constructible<T>::value, MethodStubbingProgress<R, arglist...>&>::type
    	Return(R&& r) {
```

Reply:
By "currently" you mean the code in the PR (so, without the changes in your message) or the code once you apply the changes in your message?

Reply:
I mean, if I try this "suggested change" to the PR, then it fails for native values (maybe because of other overloads)

Reply:
Using this
```        template<typename U = R>
        typename std::enable_if<!std::is_reference<T>::value && std::is_copy_constructible<T>::value, MethodStubbingProgress<R, arglist...>&>::type
    	Return(R&& r) {
```

I get this error message:

>referece_types_tests.cpp
>c:\devel\git\fakeit\tests\referece_types_tests.cpp(196): error C2664: 'fakeit::MethodStubbingProgress<R> &fakeit::MethodStubbingProgress<R>::Return(void)': cannot convert argument 1 from 'int' to 'int &'
>        with
>        [
>            R=int &
>        ]

Not saying there isn't a way to fix it but just haven't found it yet.

Also thinking the existing overload just shouldn't support R value refs at all, so should be specific  to lvalue refs, in stead of all refs, as it will always be a dangling pointer, and hence always undefined behaviour.